### PR TITLE
fix: swap quote loading spinner

### PR DESCRIPTION
**Pull request by lynnshaoyu**
https://uniswaplabs.atlassian.net/browse/WEB-2185?atlOrigin=eyJpIjoiOTE3ZjljZDQ4NTk5NDRkNjkzYzFmMjY0ZDNmZGZhODEiLCJwIjoiaiJ9

loom: https://www.loom.com/share/061951344d9745d68cfc3c9768462750

PENDING OK from fred and annabel

### CODE REVIEW

**Comment by JFrankfurt** [comment]
loom is 404'ing for me 🤔

File: src/components/CurrencyInputPanel/FiatValue.tsx [hunk 1]
```diff
@@ -29,17 +39,33 @@ export function FiatValue({
   const p = Number(fiatValue?.toFixed())
   const visibleDecimalPlaces = p < 1.05 ? 4 : 2
 
+  useEffect(() => {
+    if (isLoading) {
+      const timeout = setTimeout(() => setShowLoadingPlaceholder(true), 200)
```
Comment (JFrankfurt) [inline src/components/CurrencyInputPanel/FiatValue.tsx:44]: possible this timeout will trigger a setState on an unmounted component--should do this https://stackoverflow.com/a/60907638

here's something similar I wrote today:
```
  useEffect(() => {
    let stale = false
    let timeoutId = 0
    if (animateBuyCryptoButtonBorder) {
      timeoutId = setTimeout(() => {
        if (stale) return
        acknowledge({ system: true })
      }, ms`2 seconds`) as unknown as number
      // as unknown as number is necessary so it's not incorrectly typed as a NodeJS.Timeout
    }
    return () => {
      stale = true
      clearTimeout(timeoutId)
    }
  }, [acknowledge, animateBuyCryptoButtonBorder])
```

File: src/components/CurrencyInputPanel/FiatValue.tsx [hunk 2]
```diff
@@ -29,17 +39,33 @@ export function FiatValue({
...
-          <MouseoverTooltip text={t`The estimated difference between the USD values of input and output amounts.`}>
-            (<Trans>{priceImpact.multiply(-1).toSignificant(3)}%</Trans>)
-          </MouseoverTooltip>
-        </span>
-      ) : null}
+      {showLoadingPlaceholder ? (
+        <FiatLoadingBubble />
+      ) : (
+        <div>
+          {fiatValue && <>${fiatValue?.toFixed(visibleDecimalPlaces, { groupSeparator: ',' })}</>}
```
Comment (JFrankfurt) [inline src/components/CurrencyInputPanel/FiatValue.tsx:58]: might be a good opportunity to use @uniswap/conedison

**Review by JFrankfurt — CHANGES_REQUESTED** [review]
see comment re: stale state flag

**Comment by lynnshaoyu** [comment]
@JFrankfurt recorded a new loom try this one: https://www.loom.com/share/af69c33418c1478c9482d1328884b71b

File: src/components/CurrencyInputPanel/FiatValue.tsx (hunk 1 above)
Comment (lynnshaoyu) [reply src/components/CurrencyInputPanel/FiatValue.tsx]: smart fix!! i was wondering about this....thank you thank you. done

File: src/components/CurrencyInputPanel/FiatValue.tsx (hunk 2 above)
Comment (lynnshaoyu) [reply src/components/CurrencyInputPanel/FiatValue.tsx]: good idea done

File: src/components/CurrencyInputPanel/FiatValue.tsx [hunk 3]
```diff
@@ -26,20 +37,34 @@ export function FiatValue({
     return theme.deprecated_red1
   }, [priceImpact, theme.deprecated_green1, theme.deprecated_red1, theme.deprecated_text3, theme.deprecated_yellow1])
 
-  const p = Number(fiatValue?.toFixed())
-  const visibleDecimalPlaces = p < 1.05 ? 4 : 2
+  useEffect(() => {
+    const stale = false
+    let timeoutId = 0
+    if (isLoading && !stale) {
+      timeoutId = setTimeout(() => setShowLoadingPlaceholder(true), 200) as unknown as number
```
Comment (JFrankfurt) [inline src/components/CurrencyInputPanel/FiatValue.tsx:44]:
```suggestion
    if (isLoading) {
      timeoutId = setTimeout(() => {
        if (!stale) setShowLoadingPlaceholder(true)
      }, 200) as unknown as number
```

**Review by JFrankfurt — CHANGES_REQUESTED** [review]

File: src/components/CurrencyInputPanel/FiatValue.tsx [hunk 4]
```diff
@@ -59,7 +61,7 @@ export function FiatValue({
             <span style={{ color: priceImpactColor }}>
               {' '}
               <MouseoverTooltip text={t`The estimated difference between the USD values of input and output amounts.`}>
-                (<Trans>{priceImpact.multiply(-1).toSignificant(3)}%</Trans>)
+                (<Trans>{formatPriceImpact(priceImpact)}%</Trans>)
```
Comment (JFrankfurt) [inline src/components/CurrencyInputPanel/FiatValue.tsx:64]: confirming this wont do a double %?

**Review by JFrankfurt — APPROVED** [review]

File: src/components/CurrencyInputPanel/FiatValue.tsx (hunk 4 above)
Comment (lynnshaoyu) [reply src/components/CurrencyInputPanel/FiatValue.tsx]: it does indeed double it...my b ty for catching

**Pull request merged**